### PR TITLE
examples: update `with-supabase` example to App Router

### PR DESCRIPTION
**Pull request by dijonmusters**
### What?

Update Next.js with Supabase example

### Why?

Existing example for Next.js with Supabase is out of date

### How?

- Rename `with-supabase-auth-db-realtime` to `with-supabase`
- Update example to use App Router
- Use Supabase Auth Helpers for Next.js to configure auth cookies

### CODE REVIEW

File: examples/with-supabase/package.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    "build": "next build",
+    "start": "next start",
+    "lint": "next lint"
+  },
+  "dependencies": {
+    "@supabase/auth-helpers-nextjs": "^0.7.2",
+    "autoprefixer": "10.4.14",
+    "eslint": "8.42.0",
+    "eslint-config-next": "13.4.5",
+    "next": "13.4.5",
```
Comment (balazsorban44) [inline examples/with-supabase/package.json:16]:
```suggestion
    "next": "latest",
```

File: examples/with-supabase/package.json [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    "dev": "next dev",
+    "build": "next build",
+    "start": "next start",
+    "lint": "next lint"
+  },
+  "dependencies": {
+    "@supabase/auth-helpers-nextjs": "^0.7.2",
+    "autoprefixer": "10.4.14",
+    "eslint": "8.42.0",
+    "eslint-config-next": "13.4.5",
```
Comment (balazsorban44) [inline examples/with-supabase/package.json:15]:
```suggestion
```

File: examples/with-supabase/package.json [hunk 3]
```diff
@@ -0,0 +1,25 @@
+{
+  "name": "with-supabase",
+  "version": "0.1.0",
+  "private": true,
+  "scripts": {
+    "dev": "next dev",
+    "build": "next build",
+    "start": "next start",
+    "lint": "next lint"
```
Comment (balazsorban44) [inline examples/with-supabase/package.json:9]:
```suggestion
    "start": "next start"
```

File: examples/with-supabase/tailwind.config.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+/** @type {import('tailwindcss').Config} */
+module.exports = {
+  content: [
+    './pages/**/*.{js,ts,jsx,tsx,mdx}',
+    './components/**/*.{js,ts,jsx,tsx,mdx}',
```
Comment (balazsorban44) [inline examples/with-supabase/tailwind.config.js:5]:
```suggestion
```

**Review by balazsorban44 — DISMISSED** [review]
Thanks, left some small feedback to adhere to https://github.com/vercel/next.js/blob/canary/contributing/examples/adding-examples.md

In addition, could you also convert the example to TypeScript? Thanks!

File: examples/with-supabase/package.json [hunk 5]
```diff
@@ -0,0 +1,22 @@
+{
+  "name": "with-supabase",
+  "version": "0.1.0",
```
Comment (balazsorban44) [inline examples/with-supabase/package.json:3]:
```suggestion
```

**Review by viniciusanchieta — APPROVED** [review]

**Review by leerob — COMMENTED** [review]
Could we use TypeScript in this example please?

**Comment by dijonmusters** [comment]
@balazsorban44 and @leerob thanks for the review 🙌 

Example is now TS 👍

File: examples/with-supabase/tailwind.config.js [hunk 6]
```diff
@@ -0,0 +1,14 @@
+/** @type {import('tailwindcss').Config} */
+module.exports = {
+  content: ['./app/**/*.{js,ts,jsx,tsx,mdx}'],
+  theme: {
+    extend: {
+      backgroundImage: {
+        'gradient-radial': 'radial-gradient(var(--tw-gradient-stops))',
```
Comment (leerob) [inline examples/with-supabase/tailwind.config.js:7]: Is this still needed?

**Review by leerob — COMMENTED** [review]
Is there a deploy example I can test this out?

**Comment by dijonmusters** [comment]
> Is there a deploy example I can test this out?

https://with-supabase.vercel.app

File: examples/with-supabase/tailwind.config.js (hunk 6 above)
Comment (dijonmusters) [reply examples/with-supabase/tailwind.config.js]: Good catch! Have removed 👍

**Review by leerob — APPROVED** [review]

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: 0c0389fbd2c1121e340eb25144bfcaa88e9b5a7b

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: 0c0389fbd2c1121e340eb25144bfcaa88e9b5a7b

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

**Review by ijjk — APPROVED** [review]

**Review by ijjk — APPROVED** [review]

**Pull request merged**